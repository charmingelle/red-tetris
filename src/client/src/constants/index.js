export const LEFT = 'ArrowLeft';
export const RIGHT = 'ArrowRight';
export const DOWN = 'ArrowDown';
export const UP = 'ArrowUp';

export const UPDATE_MY_ID = 'UPDATE_MY_ID';
export const LOAD_PEOPLE = 'LOAD_PEOPLE';
export const LOAD_ROOMS = 'LOAD_ROOMS';
export const LOAD_ROOM = 'LOAD_ROOM';
export const SET_TETRO = 'SET_TETRO';
export const MOVE_TETRO = 'MOVE_TETRO';
export const MOVE_TETRO_DOWN = 'MOVE_TETRO_DOWN';
export const DROP_TETRO = 'DROP_TETRO';
export const ROTATE_TETR0 = 'ROTATE_TETRO';
export const UPDATE_MY_ROOM_ID = 'UPDATE_MY_ROOM_ID';

export const RIGHT_LIMIT = 10;
export const BOTTOM_LIMIT = 23;
export const INVISIBLE_ROW_AMOUNT = 3;
export const SQUARE_WIDTH = 40;
export const OTHER_SQUARE_WIDTH = 8;
export const WIDTH_IN_SQUARES = 10;
export const HEIGHT_IN_SQUARES = 20;

export const PENALTY_COLOR = {
  main: '#666666',
  lighter: '#999',
  darker: '#333',
};
