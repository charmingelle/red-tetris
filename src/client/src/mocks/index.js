export const RED = {
  main: '#c00',
  lighter: '#f00',
  darker: '#980000',
};

export const LINE = [
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
];

export const GI = [
  [0, 0, 0],
  [1, 1, 1],
  [0, 0, 1],
];

export const EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
];

export const NOT_EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const GAME_OVER_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, RED, RED, RED, RED, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const PILE_WITH_ROWS_TO_REMOVE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const TETRO = {
  figure: [GI],
  row: 5,
  col: 4,
  color: RED,
};

export const ROOM = {
  id: 'Room',
  leader: 'id-0',
  players: {
    'id-0': {
      id: 'id-0',
      name: 'Anna',
      pile: EMPTY_PILE,
      tetro: null,
      score: 0,
      isGameOver: false,
    },
  },
};

export const EMPTY_STORE = {
  socket: null,
  myId: null,
  myName: null,
  myRoom: null,
  people: {},
  rooms: {},
  tetro: null,
};

export const PERSON_ANNA = { name: 'Anna', score: 0 };

export const PERSON_OLYA = { name: 'Olya', score: 0 };
