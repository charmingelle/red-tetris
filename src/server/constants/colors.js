exports.RED = {
  main: '#c00',
  lighter: '#f00',
  darker: '#980000',
};

exports.ORANGE = {
  main: '#f77700',
  lighter: '#f7923a',
  darker: '#9b4e00',
};

exports.YELLOW = {
  main: '#cccc00',
  lighter: '#feff00',
  darker: '#9a9800',
};

exports.GREEN = {
  main: '#129c0a',
  lighter: '#3bd652',
  darker: '#077001',
};

exports.BLUE = {
  main: '#06cccc',
  lighter: '#0cffff',
  darker: '#108d8e',
};

exports.PURPLE = {
  main: '#0d05f3',
  lighter: '#3033f2',
  darker: '#020279',
};

exports.VIOLET = {
  main: '#9901cc',
  lighter: '#cc02ff',
  darker: '#660099',
};
