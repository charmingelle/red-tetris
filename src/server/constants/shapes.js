exports.LINE = {
  shape: [
    [0, 0, 1, 0],
    [0, 0, 1, 0],
    [0, 0, 1, 0],
    [0, 0, 1, 0],
  ],
};

exports.GI = {
  shape: [
    [0, 0, 0],
    [1, 1, 1],
    [0, 0, 1],
  ],
  initRow: 1,
};

exports.GI_MIRROR = {
  shape: [
    [0, 0, 0],
    [1, 1, 1],
    [1, 0, 0],
  ],
  initRow: 1,
};

exports.SQUARE = {
  shape: [
    [1, 1],
    [1, 1],
  ],
  initRow: 2,
};

exports.ZI = {
  shape: [
    [0, 0, 0],
    [1, 1, 0],
    [0, 1, 1],
  ],
  initRow: 1,
};

exports.ZI_MIRROR = {
  shape: [
    [0, 0, 0],
    [0, 1, 1],
    [1, 1, 0],
  ],
  initRow: 1,
};

exports.TI = {
  shape: [
    [0, 0, 0],
    [1, 1, 1],
    [0, 1, 0],
  ],
  initRow: 1,
};

exports.PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
];
