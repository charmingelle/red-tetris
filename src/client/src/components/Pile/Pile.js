import React from 'react';
import { connect } from 'react-redux';
import { Square } from '../Square';
import { SHIFT } from '../../constants';

const InnerPile = props => (
  <>
    {props.pile.map((row, rowIndex) =>
      row.map((el, colIndex) =>
        el !== 0 ? (
          <Square
            left={colIndex}
            top={rowIndex}
            color={el}
            key={`${rowIndex}+${colIndex}`}
            shift={SHIFT}
          />
        ) : null,
      ),
    )}
  </>
);

const mapStateToProps = ({ myId, myRoomId, rooms }) => ({
  pile: rooms[myRoomId].players[myId].pile,
});

export const Pile = connect(mapStateToProps)(InnerPile);
