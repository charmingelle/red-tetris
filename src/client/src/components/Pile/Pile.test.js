import React from 'react';
import { Pile } from './Pile';
import renderer from 'react-test-renderer';
import { SQUARE_WIDTH, OTHER_SQUARE_WIDTH } from '../../constants';
import { EMPTY_PILE, NOT_EMPTY_PILE } from '../../mocks';

test('Empty pile', () => {
  expect(
    renderer
      .create(<Pile pile={EMPTY_PILE} squareWidth={SQUARE_WIDTH} />)
      .toJSON(),
  ).toMatchSnapshot();
});

test('Not empty pile', () => {
  expect(
    renderer
      .create(<Pile pile={NOT_EMPTY_PILE} squareWidth={OTHER_SQUARE_WIDTH} />)
      .toJSON(),
  ).toMatchSnapshot();
});
