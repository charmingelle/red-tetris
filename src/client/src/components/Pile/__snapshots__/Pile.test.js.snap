// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty pile 1`] = `null`;

exports[`Not empty pile 1`] = `
Array [
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#c00",
        "borderBottom": "1px solid #980000",
        "borderLeft": "1px solid #f00",
        "borderRight": "1px solid #980000",
        "borderTop": "1px solid  #f00",
        "height": "8px",
        "left": "0px",
        "top": "144px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#c00",
        "borderBottom": "1px solid #980000",
        "borderLeft": "1px solid #f00",
        "borderRight": "1px solid #980000",
        "borderTop": "1px solid  #f00",
        "height": "8px",
        "left": "8px",
        "top": "144px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#c00",
        "borderBottom": "1px solid #980000",
        "borderLeft": "1px solid #f00",
        "borderRight": "1px solid #980000",
        "borderTop": "1px solid  #f00",
        "height": "8px",
        "left": "16px",
        "top": "144px",
        "width": "8px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#c00",
        "borderBottom": "1px solid #980000",
        "borderLeft": "1px solid #f00",
        "borderRight": "1px solid #980000",
        "borderTop": "1px solid  #f00",
        "height": "8px",
        "left": "16px",
        "top": "152px",
        "width": "8px",
      }
    }
  />,
]
`;
